pennylane/ops/functions/generator.py

```diff
@@ -169,8 +169,12 @@ def generator(op: qml.operation.Operator, format="prefactor"):
     >>> op = qml.RX(0.2, wires=0)
     >>> qml.generator(op, format="prefactor")  # output will always be (obs, prefactor)
     (X(0), -0.5)
-    >>> qml.generator(op, format="hamiltonian")  # output will always be a Hamiltonian
-    (-0.5) [X0]
+    >>> qml.generator(op, format="hamiltonian")  # output will always be a Hamiltonian/LinearCombination
+    -0.5 * X(0)
+    >>> with qml.operation.disable_new_opmath_cm():
+    ...     gen = qml.generator(op, format="hamiltonian")) # legacy Hamiltonian class
+    ...     print(gen, type(gen))
+    (-0.5) [X0] <class 'pennylane.ops.qubit.hamiltonian.Hamiltonian'>
```

Comment:
I guess the motivation to go into the weeds a bit more here is because `generator()` is a bit more developer facing?

Reply:
That and I found it worth pointing out since the kwarg `format = "hamiltonian"` does not give a `Hamiltonian` anymore now